File: 1. Fundamentals/removeNthFromEnd.js
const removeNthFromEnd = (head, n) => {
  var right = head,
    left = head,
    before = head;
  left = before = { next: head };

  while (n--) right = right.next;
  while (right) {
    right = right.next;
    left = left.next;
  }
  left.next = left.next.next;

  return before.next;
};

File: 1. Fundamentals/remove_nth_from_end.py
def removeNthFromEnd(head: ListNode, n: int) -> ListNode:
  right = head
  before = left = ListNode(-1)
  before.next = head
  while(right is not None and n != 0):
    n -= 1
    right = right.next
  while(right):
    right = right.next
    left = left.next
  left.next = left.next.next
  return before.next
